pkg/apis/pipeline/v1/taskrun_types.go

```diff
@@ -212,6 +212,10 @@ const (
 	TaskRunReasonResolvingStepActionRef = "ResolvingStepActionRef"
 	// TaskRunReasonImagePullFailed is the reason set when the step of a task fails due to image not being pulled
 	TaskRunReasonImagePullFailed TaskRunReason = "TaskRunImagePullFailed"
+	// TaskRunReasonCreateContainerConfigError is the reason set when the step of a task fails due to config error (e.g., missing ConfigMap or Secret)
+	TaskRunReasonCreateContainerConfigError TaskRunReason = "CreateContainerConfigError"
+	// TaskRunReasonPodCreationFailed is the reason set when the pod backing the TaskRun fails to be created (e.g., CreateContainerError)
+	TaskRunReasonPodCreationFailed TaskRunReason = "PodCreationFailed"
```

Comment:
Is `TaskRunReasonCreateContainerConfigError` a subset of `TaskRunReasonPodCreationFailed`? Since `CreateContainerError` is listed as the example here I'm hoping to understand the difference between these two failure modes better

TODO: this might become obvious as I continue to review